tests/tracer/test_sampler.py

```diff
@@ -124,19 +120,17 @@ def test_sample_rate_deviation(self):
     def test_deterministic_behavior(self):
         """Test that for a given trace ID, the result is always the same"""
         tracer = DummyTracer()
-
-        tracer._sampler = RateSampler(0.5)
+        tracer._sampler = DatadogSampler(default_sample_rate=0.5, rate_limit=-1)
```

Comment:
```suggestion
        # Since RateSampler does not set the sampling priority on a span, we will use a DatadogSampler
        tracer._sampler = DatadogSampler(default_sample_rate=0.5, rate_limit=-1)
```